pkg/function/backups_info.go

```diff
@@ -0,0 +1,166 @@
+	podFunc := BackupsInfoPodFunc(cli, tp, namespace, encryptionKey, targetPaths)
+	return pr.Run(ctx, podFunc)
+}
+
+func BackupsInfoPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPath string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to be ready", pod.Name)
+		}
+		pw, err := getPodWriter(cli, ctx, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, tp.Profile)
+		if err != nil {
+			return nil, err
+		}
+		defer cleanUpCredsFile(ctx, pw, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		snapshotIDs, err := restic.GetSnapshotIDs(tp.Profile, cli, targetPath, encryptionKey, namespace, pod.Name, pod.Spec.Containers[0].Name)
+		if err != nil {
+			if strings.Contains(err.Error(), restic.PasswordIncorrect) {
+				return map[string]interface{}{
+						BackupsInfoSnapshotIDs:       nil,
+						BackupsInfoFileCount:         nil,
+						BackupsInfoSize:              nil,
+						BackupsInfoPasswordIncorrect: "true",
+						BackupsInfoRepoUnavailable:   nil,
+					},
+					nil
+			}
+			if strings.Contains(err.Error(), restic.RepoDoesNotExist) {
+				return map[string]interface{}{
+						BackupsInfoSnapshotIDs:       nil,
+						BackupsInfoFileCount:         nil,
+						BackupsInfoSize:              nil,
+						BackupsInfoPasswordIncorrect: nil,
+						BackupsInfoRepoUnavailable:   "true",
+					},
+					nil
+			}
+			return nil, err
+		}
+		cmd, err := restic.StatsCommandByID(tp.Profile, targetPath, "" /* get all snapshot stats */, DefaultStatsMode, encryptionKey)
+		if err != nil {
+			return nil, err
+		}
+		stdout, stderr, err := kube.Exec(cli, namespace, pod.Name, pod.Spec.Containers[0].Name, cmd, nil)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stdout)
+		format.Log(pod.Name, pod.Spec.Containers[0].Name, stderr)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Failed to get backup stats")
+		}
+		// Get File Count and Size from Stats
+		_, fc, size := restic.SnapshotStatsFromStatsLog(stdout)
+		if fc == "" || size == "" {
+			return nil, errors.New("Failed to parse snapshot stats from logs")
+		}
+		return map[string]interface{}{
+				BackupsInfoSnapshotIDs:       snapshotIDs,
+				BackupsInfoFileCount:         fc,
+				BackupsInfoSize:              size,
+				BackupsInfoPasswordIncorrect: "false",
+				BackupsInfoRepoUnavailable:   "false",
+			},
+			nil
+	}
+}
+
+func (*BackupsInfoFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+	var namespace, getBackupsInfoArtifactPrefix, encryptionKey string
+	var err error
+	if err = Arg(args, BackupsInfoNamespaceArg, &namespace); err != nil {
```

Comment:
This should not be required

Reply:
Is this the correct call to get `controller ns`?
https://github.com/kanisterio/kanister/blob/6911c757d78870e04ff7120e6ca0bf28d8470290/pkg/kube/client.go#L36

Reply:
No - I believe that returns the namespace set in the `current-context` in kube config.

`kube.GetControllerNamespace()` is the right method - see it's usage in `kube_task.go`

Reply:
👍 Missed that. Thanks!